diff --git a/DEPS b/DEPS
index 57bf10484..549686199 100644
--- a/DEPS
+++ b/DEPS
@@ -7,8 +7,6 @@ use_relative_paths = True
 
 gclient_gn_args_file = 'build/config/gclient_args.gni'
 
-git_dependencies = "SYNC"
-
 gclient_gn_args = [
   'checkout_angle_internal',
   'checkout_angle_mesa',
@@ -5540,15 +5538,6 @@ hooks = [
     'condition': 'not build_with_chromium',
   },
 
-  {
-    # Update LASTCHANGE.
-    'name': 'lastchange',
-    'pattern': '.',
-    'condition': 'not build_with_chromium',
-    'action': ['python3', 'build/util/lastchange.py',
-               '-o', 'build/util/LASTCHANGE'],
-  },
-
   # Pull rc binaries using checked-in hashes.
   {
     'name': 'rc_win',
