diff --git a/scripts/update.py b/scripts/update.py
index 7e63df4..97f228d 100755
--- a/scripts/update.py
+++ b/scripts/update.py
@@ -30,6 +30,7 @@ import urllib.request
 import urllib.error
 import zipfile
 import zlib
+import certifi
 
 
 # Do NOT CHANGE this if you don't know what you're doing -- see
@@ -105,7 +106,7 @@ def DownloadUrl(url, output_file):
       sys.stdout.flush()
       request = urllib.request.Request(url)
       request.add_header('Accept-Encoding', 'gzip')
-      response = urllib.request.urlopen(request)
+      response = urllib.request.urlopen(request, cafile=certifi.where())
       total_size = None
       if 'Content-Length' in response.headers:
         total_size = int(response.headers['Content-Length'].strip())
